java/src/main/java/ai/onnxruntime/OnnxRuntime.java

```diff
@@ -5,175 +5,114 @@
 package ai.onnxruntime;
 
 import java.io.File;
-import java.io.FileNotFoundException;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Properties;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
-/**
- * Static loader for the JNI binding.
- */
+/** Static loader for the JNI binding. No public API, but called from various classes in this package to ensure shared libraries are properly loaded. */
 final class OnnxRuntime {
-    private static final Logger logger = Logger.getLogger(OnnxRuntime.class.getName());
-
-    // The initial release of the ORT API.
-    private static final int ORT_API_VERSION_1 = 1;
+  private static final Logger logger = Logger.getLogger(OnnxRuntime.class.getName());
 
-    /**
-     * Turns on debug logging during library loading.
-     */
-    public static final String LIBRARY_LOAD_LOGGING = "ORT_LOAD_LOGGING";
+  // The initial release of the ORT API.
+  private static final int ORT_API_VERSION_1 = 1;
 
-    /**
-     * Specifies that the libraries should be loaded from java.library.path rather than unzipped from the jar file.
-     */
-    public static final String LOAD_LIBRARY_PATH = "ORT_LOAD_FROM_LIBRARY_PATH";
+  /** The short name of the ONNX runtime shared library */
+  static final String ONNXRUNTIME_LIBRARY_NAME = "onnxruntime";
+  /** The short name of the ONNX runtime JNI shared library */
+  static final String ONNXRUNTIME_JNI_LIBRARY_NAME = "onnxruntime4j_jni";
 
-    private static boolean loaded = false;
+  private static boolean loaded = false;
 
-    /**
-     * The API handle.
-     */
-    static long ortApiHandle;
+  /** The API handle. */
+  static long ortApiHandle;
 
-    /**
-     * Library names stored in the jar.
-     */
-    private static final List<String> libraryNames = Arrays.asList("onnxruntime","onnxruntime4j_jni");
+  private OnnxRuntime() {}
 
-    private OnnxRuntime() {}
+  /**
+   * Loads the native C library.
+   *
+   * @throws IOException If it can't write to disk to copy out the library from the jar file.
+   */
+  static synchronized void init() throws IOException {
+    if (loaded) {
+      return;
+    }
+    Path tempDirectory = Files.createTempDirectory("onnxruntime-java");
+    tempDirectory.toFile().deleteOnExit();
```

Comment:
Minor: deleteOnClose is usually a catch-all operation, I think we should try and delete it explicitly as well after loading instead of leaving the files available till the JVM shuts down (same for the load method below)

Reply:
@Craigacp was there some nuance about closing in Windows or some OS that keeps the library open and does not like it being deleted. In an earlier version, I did this 
```java
if(!file.delete()){
 file.deleteOnExit();
}
```
the condition would fail if the OS still has the file open and it would be cleaned up on exit. What do y'all think? Netty's loader does something similar to this: https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java#L234

Reply:
I think that would be fine. I tend to just mark things `deleteOnExit` rather than trying to delete it first when I do this in other Java code, but I'm not aware of a specific issue. If Netty do that it should be fine.

Reply:
ok sounds good. I've updated it tested in linux.